src/graph.py

```diff
@@ -43,7 +72,7 @@ def remove_edge_collection(self, edge_collection: Iterable[Edge]):
             self.remove_edge(edge.u, edge.v)
 
     def generate_subgraphs_isomorphic_with(self, other: 'Graph') -> Iterable[GraphMapping]:
-        gm = nx.isomorphism.GraphMatcher(self._graph, other.nx_graph)
+        gm = nx.isomorphism.GraphMatcher(self._graph, other.nx_graph, node_match=node_equality, edge_match=edge_equality)
         return gm.subgraph_isomorphisms_iter()
```

Comment:
nasty nit: this shouldn't be .subgraph_isomorphisms_iter, but .subgraph_monomorphisms_iter. See commit "graph.py: Change isomorphism to monomorphism" in my PR for further explanation

Reply:
I see why we need monomorphism there, but let's fix that in dedicated PR